libs/ml/include/ml/optimisation/adagrad_optimiser.hpp

```diff
@@ -54,11 +55,11 @@ class AdaGradOptimiser : public Optimiser<T, C>
 };
 
 template <class T, class C>
-AdaGradOptimiser<T, C>::AdaGradOptimiser(std::shared_ptr<Graph<T>> graph,
-                                         std::string const &       input_node_name,
-                                         std::string const &       output_node_name,
+AdaGradOptimiser<T, C>::AdaGradOptimiser(std::shared_ptr<Graph<T>>       graph,
+                                         std::vector<std::string> const &input_node_names,
+                                         std::string const &             output_node_name,
                                          DataType const &learning_rate, DataType const &epsilon)
```

Comment:
pass primitives by value

Reply:
It makes sense if DataType is FixedPoint.